docs/development-update-22.44.md

```diff
@@ -0,0 +1,120 @@
+# Development Update 22.44
+
+## Overview
+
+This is an interim update ("code drop") to give the validation team
+an opportunity to take the DPDK code out for a test drive.
+
+## Project Name
+
+Note that the P4-OVS Split Architecture has a new name: **P4 Control Plane**.
+
+## What's Changed
+
+### Highlights
+
+- Restructured Stratum to support TDI variants
+- Integrated DPDK support from P4-OVS
+- Disabled unsupported port attributes in DPDK
+- Updated OVS to newer release
+- Replace `gnmi_cli` with `gnmi-ctl` in DPDK build
+
+### TDI variants
+
+When we compared the Stratum files in P4-OVS with their Tofino equivalents,
+we found significant changes in the ChassisManager, Hal, Main, and Switch
+classes, as well as SdeWrapper and YangParseTreePaths.
+
+We addressed the first four classes by creating *dpdk* and *tofino*
+subdirectories under the *tdi* directories, and giving each target its
+own implementation of the class.
+
+For SdeWrapper, we refactored `tdi_sde_wrapper.cc` (\~3800 lines) into
+multiple, smaller source files, isolating the target-specific methods
+in a single file. We then created separate versions of this file for
+the DPDK and Tofino targets, and moved them to the target-specific
+subdirectories
+
+For YangParseTreePaths, we refactored `yang_parse_tree_paths.cc`
+(originally \~4100 lines, increasing to \~5600 lines in P4-OVS) into
+multiple, smaller source files, separating out sections of code that
+had been heavily modified. We then created alternate versions for DPDK.
+See `refactoring_yang_parse_tree_paths.md` in the `stratum/docs`
+directory for details.
+
+This restructuring allowed us to port DPDK code from P4-OVS with minimal
+impact on the existing Tofino code.
+
+### DPDK support
+
+The P4 Control Plane software can now be built for either Tofino or
+DPDK. Features implemented by Stratum (P4Runtime and OpenConfig) should
+be on par with P4-OVS.
+
+Additional development work is required to support the kernel monitor
+and to allow OVS to communicate with *infraprd*.
+
+To build DPDK under CMake, you must first set the `SDE_INSTALL`
+environment variable to point to the root of the P4-DPDK install tree.
+Then, from the top-level directory of the networking recipe, enter
+
+```bash
+./make-all --target=dpdk
+```
+
+to perform the build.
+
+If you are familiar with conventional Stratum builds, you can build the
```

Comment:
Suggestion: Maybe add a heading here: For Stratum bazel builds to run unit tests? and specify command to run unit tests.

Reply:
This isn't a set of detailed instructions for validation; it's a cue for people like Max, who are already familiar with Stratum and Bazel. The key pieces of information are:

1. Stratum requires that `SDE_INSTALL` or `DPDK_INSTALL` be defined.
2. The build target to specify is in the `bin/tdi/<target>` directory.
3. You need to specify `--define target=<target>` in addition to specifying the build target.

From this, the Stratum-savvy user can figure out how to run the unit tests.

We want the ordinary user to do the CMake build. That's the only way they'll get the full package neatly served up in an install tree.

The Bazel information ultimately belongs in another document.